### ISSUE TITLE
Fields for Images + Orientatation Math

### ISSUE THREAD

**Issue by jessicagood** [StraboSpot/strabo-mobile#302]
Lat and long have been added to individual images already but what about the rest of the image properties, specifically what do we want for the Orientation for images - including which fields and what is the math? See the KoBo form for Images Properties.

As a reference the math for strike, dip, etc is https://github.com/StraboSpot/strabo-mobile/blob/master/www/app/spot/orientation-data-tab.controller.js#L61.

**Comment by panamint55** [comment]
I will try to get this math done soon.  Remind me if I drop the ball on this.  It is repurposing the S&D math.

**Comment by jessicagood** [comment]
From @panamint55 Here is the math.

1) g = sqrt (x^2 + y^2 +z^2)

2) Direction of y = diry = magnetic heading

3) Look direction = ldir

4) Look angle=alpha. Looking up, alpha +, looking down, alpha –

alpha = -(z/abs(z)) * inverse-sin(abs(z)/g)

For look direction, we take the direction the camera is facing assuming it is not a selfie.

4 cases for look direction:

1) Abs(y/x)&lt;0.5 and x&gt;0, ldir = diry+90

2) Abs(y/x)&lt;0.5 and x&lt;0, ldir = diry-90

3) Abs(y/x)&gt;0.5 and y&gt;0, ldir = diry

4) Abs(y/x)&gt;0.5 and y&lt;0, ldir = diry+180

For all of these we need to have mod(360) applied.

**Comment by jessicagood** [comment]
The major problem with implementing this is that functionality is handed over to the device camera and when it gets back to app (at which point the device motion and acceleration can be recorded) the user isn't necessarily holding the device at the same orientation as when the picture was taken.

Looked into switching to the camera plugin https://github.com/remoorejr/cordova-plugin-camera-with-exif but it doesn't look like orientation (in this case the direction and angle at which the device is held) is recorded with the EXIF data.

Putting this issue on hold.

**Comment by jasonash** [comment]
For future reference, here is the EXIF data recorded by iOS (iPad Mini 4):

Exif Image Size	3,264 × 2,448
Make	Apple
Camera Model Name	iPad mini 4
Orientation	Rotate 90 CW
Software	10.1.1
Modify Date	2016:11:17 12:51:03
20 hours, 41 minutes, 42 seconds ago
Y Cb Cr Positioning	Centered
Exposure Time	1/15
F Number	2.40
Exposure Program	Program AE
ISO	640
Exif Version	0221
Date/Time Original	2016:11:17 12:51:03
20 hours, 41 minutes, 42 seconds ago
Create Date	2016:11:17 12:51:03
20 hours, 41 minutes, 42 seconds ago
Components Configuration	Y, Cb, Cr, -
Shutter Speed Value	1/15
Aperture Value	2.40
Brightness Value	-0.9758626551
Exposure Compensation	0
Metering Mode	Multi-segment
Flash	No flash function
Focal Length	3.3 mm
Subject Area	1631 1223 1795 1077
Maker Note Apple	(838 bytes binary data)
Sub Sec Time Original	341
Sub Sec Time Digitized	341
Flashpix Version	0100
Color Space	sRGB
Sensing Method	One-chip color area
Scene Type	Directly photographed
Exposure Mode	Auto
White Balance	Auto
Focal Length In 35mm Format	31 mm
Scene Capture Type	Standard
Lens Info	3.3mm f/2.4
Lens Make	Apple
Lens Model	iPad mini 4 back camera 3.3mm f/2.4
GPS Latitude Ref	North
GPS Latitude	38.576769 degrees
GPS Longitude Ref	West
GPS Longitude	97.679542 degrees
GPS Altitude Ref	Above Sea Level
GPS Altitude	426.9754902 m
GPS Time Stamp	18:51:02.58
GPS Speed Ref	km/h
GPS Speed	0
GPS Date Stamp	2016:11:17
1 day, 9 hours, 32 minutes, 45 seconds ago
GPS Horizontal Positioning Error	10 m
Compression	JPEG (old-style)
Resolution	72 pixels/inch
Thumbnail Length	12,237

**Comment by zmichels** [comment]
Right. So the Exif doesn't include the accelerometer (direction of phone) data... just the GPS.

On Nov 18, 2016, at 11:38 AM, Jason Ash <notifications@github.com<mailto:notifications@github.com>> wrote:


For future reference, here is the EXIF data recorded by iOS (iPad Mini 4):

Exif Image Size 3,264 ? 2,448
Make Apple
Camera Model Name iPad mini 4
Orientation Rotate 90 CW
Software 10.1.1
Modify Date 2016:11:17 12:51:03
20 hours, 41 minutes, 42 seconds ago
Y Cb Cr Positioning Centered
Exposure Time 1/15
F Number 2.40
Exposure Program Program AE
ISO 640
Exif Version 0221
Date/Time Original 2016:11:17 12:51:03
20 hours, 41 minutes, 42 seconds ago
Create Date 2016:11:17 12:51:03
20 hours, 41 minutes, 42 seconds ago
Components Configuration Y, Cb, Cr, -
Shutter Speed Value 1/15
Aperture Value 2.40
Brightness Value -0.9758626551
Exposure Compensation 0
Metering Mode Multi-segment
Flash No flash function
Focal Length 3.3 mm
Subject Area 1631 1223 1795 1077
Maker Note Apple (838 bytes binary data)
Sub Sec Time Original 341
Sub Sec Time Digitized 341
Flashpix Version 0100
Color Space sRGB
Sensing Method One-chip color area
Scene Type Directly photographed
Exposure Mode Auto
White Balance Auto
Focal Length In 35mm Format 31 mm
Scene Capture Type Standard
Lens Info 3.3mm f/2.4
Lens Make Apple
Lens Model iPad mini 4 back camera 3.3mm f/2.4
GPS Latitude Ref North
GPS Latitude 38.576769 degrees
GPS Longitude Ref West
GPS Longitude 97.679542 degrees
GPS Altitude Ref Above Sea Level
GPS Altitude 426.9754902 m
GPS Time Stamp 18:51:02.58
GPS Speed Ref km/h
GPS Speed 0
GPS Date Stamp 2016:11:17
1 day, 9 hours, 32 minutes, 45 seconds ago
GPS Horizontal Positioning Error 10 m
Compression JPEG (old-style)
Resolution 72 pixels/inch
Thumbnail Length 12,237

-
